research/sentiment_analysis/sentiment_main.py

```diff
@@ -0,0 +1,188 @@
+
+  remainder = batch_size % num_gpus
+  if remainder:
+    err = ("When running with multiple GPUs, batch size "
+           "must be a multiple of the number of available GPUs. Found {} "
+           "GPUs with a batch size of {}; try --batch_size={} instead."
+          ).format(num_gpus, batch_size, batch_size - remainder)
+    raise ValueError(err)
+  return int(batch_size / num_gpus)
+
+def run_model(_):
+  """Run training and eval loop."""
+
+  tf.logging.info("Creating Estimator from Keras model...")
+  num_class = imdb.NUM_CLASS
+
+  keras_model = sentiment_model.CNN(
+      FLAGS.embedding_dim, FLAGS.vocabulary_size, FLAGS.sentence_length,
+      FLAGS.cnn_filters, num_class, FLAGS.dropout_rate, 0.0)
+  num_gpus = flags_core.get_num_gpus(FLAGS)
+  estimator = convert_keras_to_estimator(keras_model, num_gpus, FLAGS.model_dir)
+
+  # Create hooks that log information about the training and metric values
+  train_hooks = hooks_helper.get_train_hooks(
+      FLAGS.hooks,
+      batch_size=FLAGS.batch_size  # for ExamplesPerSecondHook
+  )
+  run_params = {
+      "batch_size": FLAGS.batch_size,
+      "train_epochs": FLAGS.train_epochs,
+  }
+  benchmark_logger = logger.get_benchmark_logger()
+  benchmark_logger.log_run_info(
+      model_name="sentiment_analysis",
+      dataset_name=FLAGS.dataset,
+      run_params=run_params,
+      test_id=FLAGS.benchmark_test_id)
+
+  # Training and evaluation cycle
+  def train_input_fn():
+    return dataset.input_fn(
+        FLAGS.dataset, True, per_device_batch_size(FLAGS.batch_size, num_gpus),
+        FLAGS.vocabulary_size, FLAGS.sentence_length, repeat=FLAGS.epochs_between_evals)
+
+  total_training_cycle = FLAGS.train_epochs // FLAGS.epochs_between_evals
+
+  for cycle_index in range(total_training_cycle):
+    tf.logging.info("Starting a training cycle: {}/{}".format(
+        cycle_index + 1, total_training_cycle))
+
+    # Train the model
+    estimator.train(input_fn=train_input_fn, hooks=train_hooks)
+
+    # Evaluate the model
+    eval_results = evaluate_model(estimator, num_gpus)
+
+    # Benchmark the evaluation results
+    benchmark_logger.log_evaluation_result(eval_results)
+    # Log the HR and NDCG results.
+    tf.logging.info(
+        "Iteration {}".format(eval_results))
+
+  # Clear the session explicitly to avoid session delete error
+  tf.keras.backend.clear_session()
+
+def main(_):
+  with logger.benchmark_context(FLAGS):
+    run_model(FLAGS)
+
+def define_flags():
+  """Add flags to run sentiment_main.py"""
+  # Add common flags
+  flags_core.define_base(export_dir=False)
+  flags_core.define_performance(
+      num_parallel_calls=False,
+      inter_op=False,
+      intra_op=False,
+      synthetic_data=False,
+      max_train_steps=False,
+      dtype=False
+  )
+  flags_core.define_benchmark()
+
+  flags.adopt_module_key_flags(flags_core)
+
+  flags_core.set_defaults(
+      model_dir=None,
+      train_epochs=30,
+      batch_size=30,
+      hooks="ProfilerHook")
```

Comment:
Why run ProfileHook by default?

Reply:
I took https://github.com/tensorflow/models/blob/master/official/recommendation/ncf_main.py#L304 as an example. Please let me know the proper value is ProfileHook is not.